Fix more static analyzer warnings in cookie code

RupinMittal opened the pull request:
#### 178858db52533be28686fc5e90aa465d9cb78656
<pre>
Fix more static analyzer warnings in cookie code
<a href="https://bugs.webkit.org/show_bug.cgi?id=295675">https://bugs.webkit.org/show_bug.cgi?id=295675</a>
<a href="https://rdar.apple.com/155477189">rdar://155477189</a>

Reviewed by Chris Dumez.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::MainThreadBridge::ensureOnMainThread):
* Source/WebCore/Modules/webaudio/AudioWorkletThread.cpp:
(WebCore::AudioWorkletThread::workerLoaderProxy const):
(WebCore::AudioWorkletThread::workerLoaderProxy): Deleted.
* Source/WebCore/Modules/webaudio/AudioWorkletThread.h:
* Source/WebCore/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/platform/Cookie.h:
* Source/WebCore/platform/network/cocoa/CookieCocoa.mm:
(WebCore::Cookie::operator NSHTTPCookie * _Nullable  const):
(WebCore::Cookie::operator== const):
(WebCore::Cookie::hash const):
(WebCore::Cookie::toProtectedNSHTTPCookie const):
* Source/WebCore/platform/network/cocoa/CookieStorageObserver.mm:
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::checkedWorkerLoaderProxy const):
* Source/WebCore/workers/WorkerOrWorkletThread.h:
* Source/WebCore/workers/WorkerThread.h:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/Shared/cf/CookieStorageUtilsCF.mm:
(WebKit::cookieStorageFromIdentifyingData):
* Source/WebKit/UIProcess/API/Cocoa/WKHTTPCookieStore.mm:
(WKHTTPCookieStoreObserver::create):
(-[WKHTTPCookieStore _protectedCookieStore]):
(-[WKHTTPCookieStore dealloc]):
(-[WKHTTPCookieStore getAllCookies:]):
(-[WKHTTPCookieStore setCookie:completionHandler:]):
(-[WKHTTPCookieStore setCookies:completionHandler:]):
(-[WKHTTPCookieStore deleteCookie:completionHandler:]):
(-[WKHTTPCookieStore addObserver:]):
(-[WKHTTPCookieStore removeObserver:]):
(-[WKHTTPCookieStore setCookiePolicy:completionHandler:]):
(-[WKHTTPCookieStore getCookiePolicy:]):
(-[WKHTTPCookieStore _getCookiesForURL:completionHandler:]):
(-[WKHTTPCookieStore _setCookieAcceptPolicy:completionHandler:]):
(-[WKHTTPCookieStore _flushCookiesToDiskWithCompletionHandler:]):
(-[WKHTTPCookieStore _setCookies:completionHandler:]):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebCookieJarCocoa.mm:
(WebKit::WebCookieJar::cookiesInPartitionedCookieStorage const):
(WebKit::WebCookieJar::setCookiesInPartitionedCookieStorage):
(WebKit::WebCookieJar::ensurePartitionedCookieStorage):
* Source/WebKit/WebProcess/WebPage/WebCookieCache.cpp:
(WebKit::WebCookieCache::cookiesForDOM):
(WebKit::WebCookieCache::setCookiesFromDOM):
(WebKit::WebCookieCache::didSetCookieFromDOM):
(WebKit::WebCookieCache::cookiesAdded):
(WebKit::WebCookieCache::cookiesDeleted):
(WebKit::WebCookieCache::clearForHost):
* Source/WebKit/WebProcess/WebPage/WebCookieCache.h:
* Source/WebKit/WebProcess/WebPage/WebCookieJar.h:

Canonical link: <a href="https://commits.webkit.org/297232@main">https://commits.webkit.org/297232@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1b608388784d1078e491ed4915353d2d3dfcae70

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110933 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30596 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21031 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116964 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61201 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112896 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31276 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39182 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84347 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113881 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24992 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99884 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64791 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24342 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18026 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60776 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94391 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18086 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119775 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37978 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28226 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93306 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38354 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96156 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93130 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23737 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38180 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15921 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33964 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37871 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43342 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37534 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40870 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39239 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/UIProcess/API/Cocoa/WKHTTPCookieStore.mm:
```diff
@@ -62,8 +62,9 @@ explicit WKHTTPCookieStoreObserver(id<WKHTTPCookieStoreObserver> observer)
-        if ([m_observer respondsToSelector:@selector(cookiesDidChangeInCookieStore:)])
-            [m_observer cookiesDidChangeInCookieStore:wrapper(cookieStore)];
+        RetainPtr observer = m_observer.get();
+        if ([observer respondsToSelector:@selector(cookiesDidChangeInCookieStore:)])
```
It looks like we might be able to remove this `respondsToSelector` call.

annevk commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebCookieJarCocoa.mm:
```diff
@@ -97,7 +97,8 @@
         return;
 
-    [ensurePartitionedCookieStorage() _setCookies:@[cookie.get()] forURL:cookieURL.createNSURL().get() mainDocumentURL:firstPartyURL.createNSURL().get() policyProperties:policyProperties(sameSiteInfo, cookieURL).get()];
+    RetainPtr partitionedCookieStorage = ensurePartitionedCookieStorage();
```
This seems needed because we can't inline `ensurePartitionedCookieStorage()`, but we could `const` `m_partitionedStorageForDOMCookies` for clarity.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1b608388784d1078e491ed4915353d2d3dfcae70. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47804

RupinMittal commented on Source/WebCore/workers/WorkerOrWorkletThread.h:
```diff
@@ -58,7 +58,8 @@ class WorkerOrWorkletThread : public SerialFunctionDispatcher {
     virtual WorkerDebuggerProxy* workerDebuggerProxy() const = 0;
-    virtual WorkerLoaderProxy* workerLoaderProxy() = 0;
+    virtual WorkerLoaderProxy* workerLoaderProxy() const = 0;
+    virtual CheckedPtr<WorkerLoaderProxy> checkedWorkerLoaderProxy() const = 0;
```
Fixed!

RupinMittal committed:
Fix more static analyzer warnings in cookie code
https://bugs.webkit.org/show_bug.cgi?id=295675
rdar://155477189

Reviewed by Chris Dumez.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::MainThreadBridge::ensureOnMainThread):
* Source/WebCore/Modules/webaudio/AudioWorkletThread.cpp:
(WebCore::AudioWorkletThread::workerLoaderProxy const):
(WebCore::AudioWorkletThread::workerLoaderProxy): Deleted.
* Source/WebCore/Modules/webaudio/AudioWorkletThread.h:
* Source/WebCore/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/platform/Cookie.h:
* Source/WebCore/platform/network/cocoa/CookieCocoa.mm:
(WebCore::Cookie::operator NSHTTPCookie * _Nullable  const):
(WebCore::Cookie::operator== const):
(WebCore::Cookie::hash const):
(WebCore::Cookie::toProtectedNSHTTPCookie const):
* Source/WebCore/platform/network/cocoa/CookieStorageObserver.mm:
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::checkedWorkerLoaderProxy const):
* Source/WebCore/workers/WorkerOrWorkletThread.h:
* Source/WebCore/workers/WorkerThread.h:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/Shared/cf/CookieStorageUtilsCF.mm:
(WebKit::cookieStorageFromIdentifyingData):
* Source/WebKit/UIProcess/API/Cocoa/WKHTTPCookieStore.mm:
(WKHTTPCookieStoreObserver::create):
(-[WKHTTPCookieStore _protectedCookieStore]):
(-[WKHTTPCookieStore dealloc]):
(-[WKHTTPCookieStore getAllCookies:]):
(-[WKHTTPCookieStore setCookie:completionHandler:]):
(-[WKHTTPCookieStore setCookies:completionHandler:]):
(-[WKHTTPCookieStore deleteCookie:completionHandler:]):
(-[WKHTTPCookieStore addObserver:]):
(-[WKHTTPCookieStore removeObserver:]):
(-[WKHTTPCookieStore setCookiePolicy:completionHandler:]):
(-[WKHTTPCookieStore getCookiePolicy:]):
(-[WKHTTPCookieStore _getCookiesForURL:completionHandler:]):
(-[WKHTTPCookieStore _setCookieAcceptPolicy:completionHandler:]):
(-[WKHTTPCookieStore _flushCookiesToDiskWithCompletionHandler:]):
(-[WKHTTPCookieStore _setCookies:completionHandler:]):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebCookieJarCocoa.mm:
(WebKit::WebCookieJar::cookiesInPartitionedCookieStorage const):
(WebKit::WebCookieJar::setCookiesInPartitionedCookieStorage):
(WebKit::WebCookieJar::ensurePartitionedCookieStorage):
* Source/WebKit/WebProcess/WebPage/WebCookieCache.cpp:
(WebKit::WebCookieCache::cookiesForDOM):
(WebKit::WebCookieCache::setCookiesFromDOM):
(WebKit::WebCookieCache::didSetCookieFromDOM):
(WebKit::WebCookieCache::cookiesAdded):
(WebKit::WebCookieCache::cookiesDeleted):
(WebKit::WebCookieCache::clearForHost):
* Source/WebKit/WebProcess/WebPage/WebCookieCache.h:
* Source/WebKit/WebProcess/WebPage/WebCookieJar.h:

Canonical link: https://commits.webkit.org/297232@main
